src/Autofac.Integration.Mef/RegistrationExtensions.cs

```diff
@@ -423,10 +427,136 @@ private static void ProcessExportDefinition(ContainerBuilder builder, Func<Expor
             }
         }
 
+        private static Func<object, LazyMemberInfo> getReflectionMemberImportDefinition;
+        private static Func<object, Lazy<ParameterInfo>> getReflectionParameterImportDefinition;
+
+        private static bool TryGetLazyType(this ContractBasedImportDefinition definition, out Type resultType, out Type lazyType)
+        {
+            // There are a couple of classes that are internal that provide us some information we can use to
+            // properlay guage if we need to do our lazy activation or not.
+            var definitionType = definition.GetType();
+            LazyMemberInfo? lazyMemberInfo = null;
+            resultType = null;
+            switch (definitionType.Name)
+            {
+                case "ReflectionMemberImportDefinition":
+                case "PartCreatorMemberImportDefinition":
+                    if (getReflectionMemberImportDefinition == null)
+                    {
+                        var def = Expression.Parameter(typeof(object), nameof(definition));
+                        getReflectionMemberImportDefinition = Expression.Lambda<Func<object, LazyMemberInfo>>(Expression.Property(Expression.Convert(def, definitionType), definitionType.GetProperty("ImportingLazyMember")), def).Compile();
+                    }
+
+                    lazyMemberInfo = getReflectionMemberImportDefinition(definition);
+                    break;
+                case "ReflectionParameterImportDefinition":
+                    if (getReflectionParameterImportDefinition == null)
+                    {
+                        var def = Expression.Parameter(typeof(object), nameof(definition));
+                        getReflectionParameterImportDefinition = Expression.Lambda<Func<object, Lazy<ParameterInfo>>>(Expression.Property(Expression.Convert(def, definitionType), definitionType.GetProperty("ImportingLazyParameter")), def).Compile();
+                    }
+
+                    resultType = getReflectionParameterImportDefinition(definition)?.Value?.ParameterType;
+                    break;
+            }
+
+            if (lazyMemberInfo.HasValue)
+            {
+                foreach (var v in lazyMemberInfo.Value.GetAccessors())
```

Comment:
```suggestion
                foreach (var accessor in lazyMemberInfo.Value.GetAccessors())
```

It may seem nitpicky, but using longer/more descriptive variable names makes it easier to figure out 20 lines later what this thing is. I'll add a few other comments like that in here.